syft/pkg/cataloger/debian/parse_deb_archive.go

```diff
@@ -0,0 +1,268 @@
+package debian
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+	"github.com/blakesmith/ar"
+	"github.com/klauspost/compress/zstd"
+	"github.com/ulikunitz/xz"
+)
+
+// parseDebArchive parses a Debian package archive (.deb) file and returns the packages it contains.
+// A .deb file is an ar archive containing three main files:
+// - debian-binary: Version of the .deb format (usually "2.0")
+// - control.tar.gz/xz/zst: Contains package metadata (control file, md5sums, conffiles)
+// - data.tar.gz/xz/zst: Contains the actual files to be installed (not processed by this cataloger)
+//
+// This function extracts and processes the control information to create package metadata.
+func parseDebArchive(_ context.Context, _ file.Resolver, _ *generic.Environment, reader file.LocationReadCloser) ([]pkg.Package, []artifact.Relationship, error) {
+	log.Debugf("parsing debian archive: %s", reader.RealPath)
+	
+	arReader := ar.NewReader(reader)
+	
+	var controlTarReader io.Reader
+	var md5sumsContent []byte
+	var conffilesContent []byte
+	
+	// Extract the control.tar.* file from the .deb archive
+	for {
+		header, err := arReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to read ar header: %w", err)
+		}
+		
+		log.Debugf("found archive entry: %s", header.Name)
+		
+		switch {
+		case strings.HasPrefix(header.Name, "control.tar"):
+			// Read the entire control.tar.* file
+			controlTarBytes, err := io.ReadAll(arReader)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to read control.tar: %w", err)
+			}
+			
+			// Determine compression type (.tar.gz, .tar.xz, .tar.zst)
+			compressionType := detectCompression(header.Name)
+			if compressionType == "" {
+				return nil, nil, fmt.Errorf("unsupported control file compression: %s", header.Name)
+			}
+			
+			log.Debugf("decompressing control.tar using %s compression", compressionType)
```

Comment:
there are a few log statements that probably aren't needed (this one, the "found debian package" one, and the "found archive entry")